pkgs/development/nim-packages/build-nim-package/default.nix

```diff
@@ -0,0 +1,56 @@
+{ lib, stdenv, nim }:
+
+let
+  nim_builder = stdenv.mkDerivation {
+    pname = "nim_builder";
+    inherit (nim) version;
+    dontUnpack = true;
+    nativeBuildInputs = [ nim ];
+    buildPhase = ''
+      cp ${./builder.nim} nim_builder.nim
+      nim c --nimcache:$TMPDIR nim_builder
+    '';
+    installPhase = ''
+      install -Dt $out/bin nim_builder
+    '';
+  };
+
+in { nativeBuildInputs ? [ ], configurePhase ? null, buildPhase ? null
+, checkPhase ? null, installPhase ? null, meta ? { }, ... }@attrs:
+
+stdenv.mkDerivation (attrs // {
+  nativeBuildInputs = [ nim nim_builder ] ++ nativeBuildInputs;
```

Comment:
maybe you can set strictDeps to true by default like buildPythonPackage and buildRustPackage do by default (at least)

Reply:
Good idea.